data/local/src/main/java/com/ttbmp/cinehub/data/local/mock/MockCinemaRepository.java

```diff
@@ -0,0 +1,67 @@
+package com.ttbmp.cinehub.data.local.mock;
+
+import com.ttbmp.cinehub.core.entity.Cinema;
+import com.ttbmp.cinehub.core.entity.Hall;
+import com.ttbmp.cinehub.core.entity.Movie;
+import com.ttbmp.cinehub.core.entity.Projection;
+import com.ttbmp.cinehub.core.repository.CinemaRepository;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Palmieri Ivan
+ */
+public class MockCinemaRepository implements CinemaRepository {
+
+    private final List<Cinema> cinemaList = new ArrayList<>();
+
+
+    @Override
+    public List<Cinema> getCinemaByMovie(Movie movie) {
+
+        /*Procedimento che recupera tutte le sale, tutte le proiezioni, e tutti i film dove film = movie*/
+        return cinemaList;
+    }
+
+    @Override
+    public List<Cinema> getAllCinemaList() {
+        this.cinemaList.addAll(Arrays.asList(
+                new Cinema("The cinema", "Via ovunque", "Tivoli"),
+                new Cinema("UCI cinema", "Via delle vie", "Monterotondo"),
+                new Cinema("Cinema Giuseppetti", "Via Nettuno", "Nettuno"),
+                new Cinema("Cinema Bous", "Via rossi", "Milano")
+        ));
+        return cinemaList;
+    }
+
+    @Override
+    public List<Cinema> getCinemaByHall(Hall hall) {
+        getAllCinemaList();
+        List<Cinema> listCinemaByHall = new ArrayList<>();
+        for (Cinema cinema : cinemaList) {
+            for (int j = 0; j < cinema.getHallList().size(); j++) {
+                if (cinema.getHallList().get(j) == hall) {
+                    listCinemaByHall.add(cinema);
+                }
+            }
+        }
+        return listCinemaByHall;
+
+    }
+
+
+    @Override
+    public List<Cinema> getCinemaByProjection(List<Projection> projectionList) {
```

Comment:
A projection is traversable to obtain the cinema, this operation is useless.

Reply:
The method is used to see in which cinemas that specific projection is located, since the projection is a datum retrieved from the database, it will have associated a cinema attribute, and this is exactly what I want to obtain